rlberry/agents/torch/sac/sac.py

```diff
@@ -0,0 +1,307 @@
+import torch
+import torch.nn as nn
+import logging
+
+import gym.spaces as spaces
+from rlberry.agents import AgentWithSimplePolicy
+from rlberry.agents.utils.memories import Memory
+from rlberry.agents.torch.utils.training import optimizer_factory
+from rlberry.agents.torch.utils.models import default_policy_net_fn
+from rlberry.agents.torch.utils.models import default_value_net_fn
+from rlberry.agents.torch.utils.models import default_twinq_net_fn
+from rlberry.utils.torch import choose_device
+from rlberry.wrappers.uncertainty_estimator_wrapper import UncertaintyEstimatorWrapper
+
+logger = logging.getLogger(__name__)
+
+
+class SACAgent(AgentWithSimplePolicy):
+    """
+    Soft Actor Critic Agent (WIP). So far it is just a copy of A2C
+
+    A2C, or Advantage Actor Critic, is a synchronous version of the A3C policy
+    gradient method. As an alternative to the asynchronous implementation of
+    A3C, A2C is a synchronous, deterministic implementation that waits for each
+    actor to finish its segment of experience before updating, averaging over
+    all of the actors. This more effectively uses GPUs due to larger batch sizes.
+
+    Parameters
+    ----------
+    env : Model
+        Online model with continuous (Box) state space and discrete actions
+    batch_size : int
+        Number of episodes to wait before updating the policy.
+    horizon : int
+        Horizon.
+    gamma : double
+        Discount factor in [0, 1].
+    entr_coef : double
+        Entropy coefficient.
+    learning_rate : double
+        Learning rate.
+    optimizer_type: str
+        Type of optimizer. 'ADAM' by defaut.
+    k_epochs : int
+        Number of epochs per update.
+    policy_net_fn : function(env, **kwargs)
+        Function that returns an instance of a policy network (pytorch).
+        If None, a default net is used.
+    value_net_fn : function(env, **kwargs)
+        Function that returns an instance of a value network (pytorch).
+        If None, a default net is used.
+    policy_net_kwargs : dict
+        kwargs for policy_net_fn
+    value_net_kwargs : dict
+        kwargs for value_net_fn
+    use_bonus : bool, default = False
+        If true, compute an 'exploration_bonus' and add it to the reward.
+        See also UncertaintyEstimatorWrapper.
+    uncertainty_estimator_kwargs : dict
+        Arguments for the UncertaintyEstimatorWrapper
+    device : str
+        Device to put the tensors on
+
+    References
+    ----------
+    Mnih, V., Badia, A.P., Mirza, M., Graves, A., Lillicrap, T., Harley, T.,
+    Silver, D. & Kavukcuoglu, K. (2016).
+    "Asynchronous methods for deep reinforcement learning."
+    In International Conference on Machine Learning (pp. 1928-1937).
```

Comment:
Maybe you can add 
```
See also :class:`rlberry.agents.torch.A2CAgent` 
```
this will make a link in the doc to A2CAgent, because the agents are related.